circle.yml

```diff
@@ -1,38 +1,151 @@
+           - node_modules
+  unit_tests:
+    machine: true
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install --upgrade pip
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Pull images and setup
+          command: |
+            set -x
+            docker-compose up -d
+            sleep 10
+            docker-compose run --rm postgres psql -h postgres -U postgres -c "create database tests;"
+      - run:
+          name: Run tests
+          command: docker-compose run --user root server tests
+      - store_artifacts:
+          path: junit.xml
+      - store_artifacts:
+          path: coverage.xml
+  integration_tests:
+    working_directory: ~/redash
+    machine: true
+    environment:
+      REDASH_SERVER_URL : "http://127.0.0.1:5000/"
+      DOCKER_IMAGE: mozilla/redash-ui-tests
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install --upgrade pip
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Pull redash images
+          command: |
+            set -x
+            docker-compose -f docker-compose.yml up -d
+            sleep 10
+      - run:
+          name: Pull redash-ui-tests
+          command: docker pull "${DOCKER_IMAGE}":latest
+      - run:
+          name: Setup redash instance
+          command: |
+            set -x
+            docker-compose run --rm --user root server create_db
+            docker-compose run --rm postgres psql -h postgres -U postgres -c "create database tests"
+            docker-compose run --rm --user root server /app/manage.py users create_root root@example.com "rootuser" --password "IAMROOT" --org default
+            docker-compose run --rm --user root server /app/manage.py ds new "ui-tests" --type "url" --options '{"title": "uitests"}'
+            docker-compose run -d -p 5000:5000 --user root server
+      - run:
+          name: Run tests
+          command: |
+            set -x
+            docker run --net="host" --env REDASH_SERVER_URL="${REDASH_SERVER_URL}" "${DOCKER_IMAGE}"
+      - store_artifacts:
+          path: report.html
+  build-deploy: # build for master or release branches
+    docker:
+      - image: circleci/node:6
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Build
+          command: bin/pack "$CIRCLE_BRANCH"
+  build-release: # build for tags
+    docker:
+      - image: circleci/node:6
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Tag deploy
+          command: |
+            set -x
+            bin/pack "$CIRCLE_TAG"
```

Comment:
Don't we need to define the tar ball file as an artifact?

Reply:
So this where I need some clarification. Do we just want the artifact as a circleci artifact? Or does it need to be something specific. Circleci 2.0 doesn't have much on the old ```tarball``` command, just stating that it was an artifact.

Reply:
Looking at current master builds on circle, there is an artifact with a ```tar.gz``` extension. Is this made with the command ```bin/pack``` ?

Reply:
@jrbenny35 Yep, bin/pack basically puts everything in the repo clone into a tarball (except a few things like git history and node_modules etc).

Reply:
Yes, it just needs to be available as a CircleCI artifact.